cwms-data-api/src/test/java/cwms/cda/api/OpenApiDocTest.java

```diff
@@ -0,0 +1,486 @@
+        List<OpenApiParamInfo> expectedQueryParameters = testInfo.getQueryParameters();
+        List<OpenApiParamInfo> expectedPathParameters = testInfo.getPathParameters();
+
+        Set<OpenApiParamUsageInfo> receivedQueryParameters = parsedParamInfo.getQueryParams();
+        Set<OpenApiParamUsageInfo> receivedPathParameters = parsedParamInfo.getPathParams();
+        OpenApiParamUsageInfo receivedResourceId = parsedParamInfo.getResourceId();
+        return () -> assertAll("Testing " + testInfo.getMethod().getName(),
+                               () -> testQueryParameters(expectedQueryParameters, receivedQueryParameters),
+                               () -> testPathParameters(expectedPathParameters, receivedPathParameters, receivedResourceId));
+    }
+
+    private void testPathParameters(List<OpenApiParamInfo> expectedPathParameters, Set<OpenApiParamUsageInfo> receivedPathParameters,
+                                    OpenApiParamUsageInfo receivedResourceId) {
+        List<OpenApiParamUsageInfo> verifiedUsages = new ArrayList<>();
+        List<OpenApiParamInfo> expectedParams = new ArrayList<>();
+        List<OpenApiParamInfo> missingItems = new ArrayList<>();
+        List<OpenApiParamUsageInfo> receivedItems = new ArrayList<>(receivedPathParameters);
+
+        if (receivedResourceId != null) {
+            //Special case, equivalent to the last expectedPathParameters, but it can have an ambiguous name.
+            if (!expectedPathParameters.isEmpty()) {
+                String name = expectedPathParameters.get(expectedPathParameters.size() - 1).getName();
+                receivedResourceId.getParamInfo().setName(name);
+            }
+
+            receivedItems.add(receivedResourceId);
+        }
+
+        for (OpenApiParamInfo paramInfo : expectedPathParameters) {
+            OpenApiParamUsageInfo equivalent = null;
+            for (OpenApiParamUsageInfo paramUsage : receivedItems) {
+                if (paramUsage.getParamInfo().getName().equals(paramInfo.getName())) {
+                    equivalent = paramUsage;
+                    break;
+                }
+            }
+            if (equivalent == null) {
+                missingItems.add(paramInfo);
+            } else {
+                receivedItems.remove(equivalent);
+                verifiedUsages.add(equivalent);
+                expectedParams.add(paramInfo);
+            }
+        }
+
+        String extraInfo = receivedItems.stream()
+                                        .map(p -> p.getParamInfo().getName())
+                                        .collect(Collectors.joining(", "));
+        String missingInfo = missingItems.stream()
+                                         .map(OpenApiParamInfo::getName)
+                                         .collect(Collectors.joining(", "));
+        assertAll(() -> assertTrue(receivedItems.isEmpty(), "Found extra path parameters: " + extraInfo),
```

Comment:
I feel like this should explain what "extra" means.

Reply:
Updated it to say it's a used undocumented path parameter.